dialogue-clients/src/main/java/com/palantir/dialogue/clients/ReloadingClientFactory.java

```diff
@@ -333,14 +311,42 @@ public String toString() {
             return "AlwaysThrowingChannel{exceptionSupplier="
                     + exceptionSupplier.get().getMessage() + '}';
         }
+
+        @Override
+        public List<Channel> perHostChannels() {
+            return Collections.emptyList();
+        }
+    }
+
+    interface InternalDialogueChannel extends Channel {
+        List<Channel> perHostChannels();
+    }
+
+    private static final class InternalDialogueChannelFromDialogueChannel implements InternalDialogueChannel {
+
+        private final DialogueChannel dialogueChannel;
```

Comment:
Not sure why we don't just use the DialogueChannel directly. Don't we want one that also implements EndpointChannelFactory?

Reply:
There might be a better way to do it, but if you have a look at getInternalDialogueChannel, we have things like AlwaysThrowingChannel, and the return type must have the same supertype.

Reply:
Got it. Might be helpful to add that detail to InternalDialogueChannel javadoc.